core/arch/arm/plat-hisilicon/a53_plat_init.S

```diff
@@ -0,0 +1,106 @@
+//  SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2019, HiSilicon Technologies Co., Ltd.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
```

Comment:
Please keep only:
```C
//  SPDX-License-Identifier: BSD-2-Clause
/*
 * Copyright (c) 2019, HiSilicon Technologies Co., Ltd.
 */
```

We don't need all the license boilerplate, because the SPDX license identifier describes the license terms unambiguously.

Reply:
I am not sure, but the only thing i want to make sure is that we have no legal issues by doing so, would you double check it for me?   Now i only see the Linaro is doing like what you suggested, but no other SOC vendors.

Reply:
There are commits in optee_os that use the SPDX identifier and no boilerplate license text. For instance, this one by NXP: https://github.com/OP-TEE/optee_os/commit/ad22421.

The Linux kernel does the same. See for instance:
- https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=09d290f0ba21
- https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=2849dd8bc72b6

This is explained at: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/license-rules.rst?h=v5.0-rc7#n40, more precisely: "An alternative to boilerplate text is the use of [SPDX] identifiers".

So I really think it is OK to not add the license text.